src/modules/ekf2/EKF2.cpp

```diff
@@ -354,12 +354,37 @@ void EKF2::Run()
 
 					// Validate the ekf origin status.
 					_ekf.getEkfGlobalOrigin(origin_time, latitude, longitude, altitude);
-					PX4_INFO("New NED origin (LLA): %3.10f, %3.10f, %4.3f\n", latitude, longitude, static_cast<double>(altitude));
+					PX4_INFO("New NED origin (LLA): %3.10f, %3.10f, %4.3f", latitude, longitude, static_cast<double>(altitude));
 				}
 			}
 		}
 	}
 
+	if (_vehicle_local_position_sub.updated()) {
+		vehicle_local_position_s lpos;
+
+		if (_vehicle_local_position_sub.update(&lpos)) {
+
+			if (lpos.ref_timestamp > 0
+			    && lpos.ref_timestamp > _ekf.global_origin().getProjectionReferenceTimestamp()
+			    && (abs(lpos.ref_lat - _ekf.global_origin().getProjectionReferenceLat()) > 1e-9
+				|| abs(lpos.ref_lon - _ekf.global_origin().getProjectionReferenceLon()) > 1e-9
+				|| abs(lpos.ref_alt - _ekf.getEkfGlobalOriginAltitude()) > 1e-3f)) {
+
+				_ekf.setEkfGlobalOrigin(lpos.ref_lat, lpos.ref_lon, lpos.ref_alt);
+				// Validate the ekf origin status.
+
+				uint64_t origin_time {};
+				double latitude, longitude;
+				float altitude;
+
+				_ekf.getEkfGlobalOrigin(origin_time, latitude, longitude, altitude);
+				PX4_INFO("New NED origin (LLA): %3.10f, %3.10f, %4.3f", latitude, longitude, static_cast<double>(altitude));
```

Comment:
Add the instance when printing?

https://github.com/PX4/PX4-Autopilot/blob/4c0bb706c7276ee2db6b01774041e2253d9beb94/src/modules/ekf2/EKF2.cpp#L427